src/editor/ui_menus/scenario_tool_road_options_menu.cc

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (C) 2002-2019 by the Widelands Development Team
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ */
+
+#include "editor/ui_menus/scenario_tool_road_options_menu.h"
+
+#include "editor/editorinteractive.h"
+
+inline EditorInteractive& ScenarioToolRoadOptionsMenu::eia() {
+	return dynamic_cast<EditorInteractive&>(*get_parent());
+}
+
+static EditorActionArgs::RoadMode roadmode(int32_t i) {
+	switch (i) {
```

Comment:
Make `EditorActionArgs::RoadMode` an enum class. Al you need to do then is a static cast. You can add an assert to this function if you want, the other function won't even need that.